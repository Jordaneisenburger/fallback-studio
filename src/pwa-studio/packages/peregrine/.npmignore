.eslintrc.js
/lib/**/__{docs,helpers,mocks,tests}__/**
/scripts
