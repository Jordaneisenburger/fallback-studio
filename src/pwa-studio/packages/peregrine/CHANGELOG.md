# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.5.0"></a>
# 0.5.0 (2018-08-13)


### Features

* multicasting REST API client ([#164](https://github.com/magento/pwa-studio/issues/164)) ([2852e14](https://github.com/magento/pwa-studio/commit/2852e14)), closes [#140](https://github.com/magento/pwa-studio/issues/140)
* Simulators for async testing ([#24](https://github.com/magento/pwa-studio/issues/24)) ([a380da9](https://github.com/magento/pwa-studio/commit/a380da9))
