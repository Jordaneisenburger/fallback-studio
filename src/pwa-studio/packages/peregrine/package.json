{
  "name": "@magento/peregrine",
  "version": "2.1.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "dist/index.js",
  "scripts": {
    "build": "concurrently --raw yarn:build:cjs yarn:build:esm",
    "build:cjs": "BABEL_ENV=production babel src --out-dir dist --root-mode 'upward' --source-maps",
    "build:dev": "yarn run build",
    "build:esm": "BABEL_ENV=development babel src --out-dir esm --root-mode 'upward' --source-maps",
    "clean": "rimraf dist esm",
    "prepublishOnly": "yarn run clean && yarn run build",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist",
    "watch": "yarn run build --watch"
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@babel/cli": "~7.2.3",
    "@babel/core": "~7.3.4",
    "@babel/plugin-proposal-class-properties": "~7.3.4",
    "@babel/plugin-proposal-object-rest-spread": "~7.3.4",
    "@babel/plugin-syntax-dynamic-import": "~7.2.0",
    "@babel/plugin-syntax-jsx": "~7.2.0",
    "@babel/plugin-transform-react-jsx": "~7.3.0",
    "@babel/plugin-transform-runtime": "~7.4.4",
    "@babel/runtime": "~7.4.2",
    "@storybook/react": "~4.1.13",
    "concurrently": "~4.1.0",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6",
    "rimraf": "~2.6.3"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "8.10.0 || >=10.7.0",
    "yarn": ">=1.12.0"
  },
  "module": "esm/index.js",
  "jsnext:main": "src/index.js",
  "es2015": "src/index.js",
  "sideEffects": false
}
