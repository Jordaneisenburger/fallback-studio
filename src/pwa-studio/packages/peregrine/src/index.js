import * as RestApi from './RestApi';
import * as Util from './util';

// hooks
export { useApolloContext } from './hooks/useApolloContext';
export { useEventListener } from './hooks/useEventListener';
export { useDropdown } from './hooks/useDropdown';
export { useQuery } from './hooks/useQuery';
export { useQueryResult } from './hooks/useQueryResult';
export { useSearchParam } from './hooks/useSearchParam';
export {
    WindowSizeContextProvider,
    useWindowSize
} from './hooks/useWindowSize';

// components
export { default as ContainerChild } from './ContainerChild';
export { default as List, Items, Item } from './List';
export { default as Page } from './Page';
export { default as Price } from './Price';
export { default as Router } from './Router';

// misc
export { RestApi };
export { Util };
export { default as createTestInstance } from './util/createTestInstance';
