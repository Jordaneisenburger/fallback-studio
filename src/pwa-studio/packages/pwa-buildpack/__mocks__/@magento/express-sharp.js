const middleware = jest.fn();
const expressSharp = (module.exports = jest.fn().mockReturnValue(middleware));
expressSharp.__mockMiddleware = middleware;
