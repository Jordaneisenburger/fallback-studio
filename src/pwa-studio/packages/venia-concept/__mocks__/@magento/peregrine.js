import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/lib');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useEventListener = jest.fn(peregrine.useEventListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);
const useWindowSize = jest.fn(peregrine.useWindowSize);
const useToasts = jest.fn(peregrine.useToasts);
// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useEventListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam,
    useWindowSize,

    useToasts
};
