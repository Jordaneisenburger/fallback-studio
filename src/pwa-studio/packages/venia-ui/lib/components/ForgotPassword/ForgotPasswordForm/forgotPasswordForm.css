.root {
    display: grid;
    gap: 1rem;
    justify-items: stretch;
}

.buttonContainer {
    align-items: center;
    display: flex;
    justify-content: center;
    margin-top: 1rem;
    text-align: center;
}
