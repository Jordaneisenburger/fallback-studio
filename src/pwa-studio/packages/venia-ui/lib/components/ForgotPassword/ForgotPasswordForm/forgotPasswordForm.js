import React from 'react';
import { func, shape, string } from 'prop-types';
import { Form } from 'informed';

import Button from '../../Button';
import Field from '../../Field';
import TextInput from '../../TextInput';

import { isRequired } from '../../../util/formValidators';

import { mergeClasses } from '../../../classify';
import defaultClasses from './forgotPasswordForm.css';

const ForgotPasswordForm = props => {
    const classes = mergeClasses(defaultClasses, props.classes);
    const { initialValues, isResettingPassword, onSubmit } = props;

    return (
        <Form
            className={classes.root}
            initialValues={initialValues}
            onSubmit={onSubmit}
        >
            <Field label="Email Address" required={true}>
                <TextInput
                    autoComplete="email"
                    field="email"
                    validate={isRequired}
                    validateOnBlur
                />
            </Field>
            <div className={classes.buttonContainer}>
                <Button
                    disabled={isResettingPassword}
                    type="submit"
                    priority="high"
                >
                    Submit
                </Button>
            </div>
        </Form>
    );
};

ForgotPasswordForm.propTypes = {
    classes: shape({
        form: string,
        buttonContainer: string
    }),
    initialValues: shape({
        email: string
    }),
    onSubmit: func.isRequired
};

ForgotPasswordForm.defaultProps = {
    initialValues: {}
};

export default ForgotPasswordForm;
