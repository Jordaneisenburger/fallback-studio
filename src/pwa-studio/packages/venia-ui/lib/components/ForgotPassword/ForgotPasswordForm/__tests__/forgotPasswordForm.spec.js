import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import { Form } from 'informed';

import ForgotPasswordForm from '../forgotPasswordForm';

test('renders correctly', () => {
    const wrapper = createTestInstance(
        <ForgotPasswordForm onSubmit={() => {}} />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('calls onSubmit callback', () => {
    const onSubmit = jest.fn();
    const wrapper = createTestInstance(
        <ForgotPasswordForm onSubmit={onSubmit} />
    );
    wrapper.root.findByType(Form).props.onSubmit();

    expect(onSubmit).toBeCalled();
});
