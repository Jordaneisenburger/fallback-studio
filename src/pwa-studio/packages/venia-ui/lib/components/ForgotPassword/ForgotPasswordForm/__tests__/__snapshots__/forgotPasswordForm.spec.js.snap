// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div>
    <label>
      <span />
      Email Address
    </label>
    <span
      style={
        Object {
          "--iconsAfter": 0,
          "--iconsBefore": 0,
        }
      }
    >
      <span>
        <input
          autoComplete="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
      </span>
      <span />
      <span />
    </span>
    <p />
  </div>
  <div>
    <button
      type="submit"
    >
      <span>
        Submit
      </span>
    </button>
  </div>
</form>
`;
